src/main/java/com/impactupgrade/nucleus/controller/AccountingController.java

```diff
@@ -4,38 +4,76 @@
 
 package com.impactupgrade.nucleus.controller;
 
+import com.impactupgrade.nucleus.entity.JobStatus;
+import com.impactupgrade.nucleus.entity.JobType;
+import com.impactupgrade.nucleus.environment.Environment;
 import com.impactupgrade.nucleus.environment.EnvironmentFactory;
+import com.impactupgrade.nucleus.service.segment.SynchronizationService;
 
 import javax.servlet.http.HttpServletRequest;
 import javax.ws.rs.Consumes;
 import javax.ws.rs.FormParam;
+import javax.ws.rs.GET;
 import javax.ws.rs.POST;
 import javax.ws.rs.Path;
 import javax.ws.rs.Produces;
+import javax.ws.rs.QueryParam;
 import javax.ws.rs.core.Context;
 import javax.ws.rs.core.MediaType;
 import javax.ws.rs.core.Response;
+import java.util.Calendar;
 
 @Path("/accounting")
 public class AccountingController {
 
-    private static final String DATE_FORMAT = "yyyy-MM-dd";
+  private static final String DATE_FORMAT = "yyyy-MM-dd";
 
-    protected final EnvironmentFactory environmentFactory;
+  protected final EnvironmentFactory environmentFactory;
 
-    public AccountingController(EnvironmentFactory environmentFactory) {
-        this.environmentFactory = environmentFactory;
+  public AccountingController(EnvironmentFactory environmentFactory) {
+    this.environmentFactory = environmentFactory;
+  }
+
+  @GET
+  @Path("/sync/contacts/daily")
+  public Response syncDaily(@QueryParam("syncDays") Integer syncDays, @Context HttpServletRequest request) throws Exception {
+    Environment env = environmentFactory.init(request);
+
+    Calendar lastSync = Calendar.getInstance();
+    // run daily, but setting this high to catch previous misses
+    if (syncDays == null || syncDays <= 0) {
+      syncDays = 3;
     }
+    lastSync.add(Calendar.DATE, -syncDays);
+
+    Runnable thread = () -> {
+      env.startJobLog(JobType.EVENT, null, "Contacts Sync: Daily", "Nucleus Portal");
+
+      SynchronizationService synchronizationService = env.synchronizationService("xeroSync");
```

Comment:
Rather than hard code Xero here, need to loop over all the SynchronizationServices like we do for CommunicationService and PaymentGatewayService. We'll need that once this starts syncing contact updates to both Stripe and Xero at once.